src/extensions/savetoLaserfiche/GetDocumentDataDialog.tsx

```diff
@@ -117,15 +115,58 @@ function GetDocumentDialogData(props: {
   ));
 
   React.useEffect(() => {
-    const libraryUrl = props.context.pageContext.list.title;
-    GetAllFieldsValues(libraryUrl, props.spFileInfo.fileId).then(
-      (allSpFieldValues: object) => {
-        getDocumentDataAsync(allSpFieldValues);
-      }
-    );
+    saveDocumentToLaserficheAsync();
```

Comment:
So what I did here is similar to what the blog post is describing. I put all of the await logic in an async function and am calling that function. This allows me to use await within the function. I can't await the top level function though because the useEffect lambda shouldn't be async.

That being said, I went down the rabbit hole of React looking into this and I think there are some cases where I should be returning a clean up function, so I am looking into how to do that.